plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -415,6 +423,209 @@ createWidget("discourse-poll-info", {
   }
 });
 
+function transformUserFieldToLabel(fieldName) {
+  let transformed = fieldName.split("_").filter(Boolean);
+  transformed[0] = classify(transformed[0]);
+  return transformed.join(" ");
+}
+
+createWidget("discourse-poll-grouped-pies", {
+  tagName: "div.poll-grouped-pies",
+  buildAttributes(attrs) {
+    return {
+      id: `poll-results-grouped-pie-charts-${attrs.id}`
+    };
+  },
+
+  html(attrs) {
+    const fields = Object.assign({}, attrs.groupableUserFields);
+    const fieldSelectId = `field-select-${attrs.id}`;
+    attrs.groupedBy = attrs.groupedBy || fields[0];
+
+    let contents = [];
+
+    contents.push(
+      h(
+        `select#${fieldSelectId}.poll-group-by-selector`,
+        { value: attrs.groupBy },
+        attrs.groupableUserFields.map(field => {
+          return h(
+            "option",
+            { value: field },
+            transformUserFieldToLabel(field)
+          );
+        })
+      )
+    );
+
+    contents.push(h("div.clearfix"));
+
+    later(() => {
```

Comment:
I'm sorry but I highly dislike this later + ajax inside html, `html` should be  a pure function more or less.

What you could do is having a `state.groupedPollResultsLoaded` and architecture your html like this:

```
init() {
  this.loadGroupedPollResults().then((results) => {
     this.state.groupedResults = results.grouped_results;
     this.state.isLoaded = true;
   });
},

html() {
  let contents = [];
  contents.push(foo);
  if (this.state.isLoaded) {
      content.push(this.attach("bar"));
  }
  return contents;
}
```

This is not complete code, but I think this should work and be cleaner.